sklearn/decomposition/tests/test_pca.py

```diff
@@ -454,6 +454,8 @@ def test_pca_zero_noise_variance_edge_cases(svd_solver):
 
```

Comment:
```suggestion
    # Ensure no zero division error for n_components == n_features < n_samples
    pca.score(X)
    
```

If we're testing one of the two cases, it might be good to test the other case too, for symmetry.